include/veriblock/blockchain/pop/fork_resolution.hpp

```diff
@@ -499,6 +499,10 @@ struct PopAwareForkResolutionComparator {
   int comparePopScore(ProtectedBlockTree& ed,
                       protected_index_t& candidate,
                       ValidationState& state) {
+    VBK_ASSERT_MSG(!candidate.finalized,
```

Comment:
If the candidate is finalized, it is part of the best chain  by definition

Reply:
ah yes! we can theoretically compare current block on main chain and other block on main chain